IBMigrationTool/main.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  main.swift
+//  IBMigrationTool
+//
+//  Created by Simon Mitchell on 17/09/2020.
+//  Copyright © 2020 threesidedcube. All rights reserved.
+//
+
+import Foundation
+
+extension FileManager {
+    
+    /// Recurses the given directory for files with a given extension
+    /// - Parameters:
+    ///   - type: The extension to find files for
+    ///   - directory: The directory to search in
+    /// - Returns: An array of file paths
+    func recursivePathsForResource(_ type: String?, directory: String) -> [String] {
+        
+        var filePaths: [String] = []
+        let enumerator = FileManager.default.enumerator(atPath: directory)
+        
+        while let element = enumerator?.nextObject() as? NSString {
+            
+            if type == nil || element.pathExtension == type {
+                filePaths.append(directory + "/" + (element as String))
+            }
+        }
+        
+        return filePaths
+    }
+}
+
+/// Converts all `UIView+Displayable` runtime attributes to use the
+/// `CALayer` equivalents in the file at the given path
+/// - Parameter path: The file path to convert
+func migrateUserDefinedRuntimeAttributesInInterfaceBuilderFile(at path: String) {
+        
+    let fileURL = URL(fileURLWithPath: path)
+    guard let data = try? Data(contentsOf: fileURL),
+          var string = String(data: data, encoding: .utf8) else { return }
+    
+    print("Migrating user defined runtime attributes in \(path)")
+    
+    // These use the same type as `CALayer` equivalents, so can be
+    // mapped simply using the same regex
+    let directlyMappableAttributes: [(String, String)] = [
+        ("color", "borderColor"),
+        ("number", "borderWidth"),
+        ("number", "cornerRadius"),
+        ("color", "shadowColor"),
+        ("number", "shadowOpacity"),
+        ("number", "shadowRadius")
+    ]
+    
+    directlyMappableAttributes.forEach { (attribute)  in
+                
+        string = string.replacingOccurrences(
+            of: "<userDefinedRuntimeAttribute(\\s+)type=\"\(attribute.0)\"(\\s+)keyPath=\"(\(attribute.1))\">",
+            with: "<userDefinedRuntimeAttribute$1type=\"\(attribute.0)\"$2keyPath=\"layer.$3\">",
+            options: .regularExpression,
+            range: nil
+        )
+    }
+    
+    // shadowOffset is a bit more complex because we used `CGPoint` but
+    // `CALayer` uses `CGSize`
+    
+    // Note this regex isn't perfect as it allows for x,y to take the form 1.2.1.3.1.2 e.t.c
+    // however this should be impossible to enter in Interface Builder and it makes replacing it
+    // far simpler so we will leave be for now!
+    string = string.replacingOccurrences(
```

Comment:
This is a long shot and defo not required if it's going to be long, but would be reeeeally useful to see number of changes per file (so we know when nothings changed etc)

Reply:
As you point out the git diff can do this for us :)